pkg/asset/installconfig/gcp/client.go

```diff
@@ -0,0 +1,74 @@
+package gcp
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	compute "google.golang.org/api/compute/v1"
+	"google.golang.org/api/option"
+)
+
+// API represents the calls made to the API.
+type API interface {
+	GetNetwork(ctx context.Context, network, project string) (*compute.Network, error)
+	GetSubnetworks(ctx context.Context, network, project, region string) ([]*compute.Subnetwork, error)
+}
+
+// Client makes calls to the GCP API.
+type Client struct{}
+
+// GetNetwork uses the GCP Compute Service API to get a network by name from a project.
+func (c *Client) GetNetwork(ctx context.Context, network, project string) (*compute.Network, error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	svc, err := getComputeService(ctx)
+	if err != nil {
+		return nil, err
+	}
+	res, err := svc.Networks.Get(project, network).Context(ctx).Do()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get network %s", network)
+	}
+	return res, nil
+}
+
+// GetSubnetworks uses the GCP Compute Service API to retrieve all subnetworks in a given network.
+func (c *Client) GetSubnetworks(ctx context.Context, network, project, region string) ([]*compute.Subnetwork, error) {
+	ctx, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+
+	svc, err := getComputeService(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	req := svc.Subnetworks.List(project, region)
+	var res []*compute.Subnetwork
+	if err := req.Pages(ctx, func(page *compute.SubnetworkList) error {
```

Comment:
the list has filters, that filter on server side, we should use then instead of client side.

Reply:
`network eq .*ci-op--45krg-network`
`svc.Subnetworks.List(project, region).Filter("network eq .*<network>")`

Reply:
I implemented this but the filter seems limited. I don't see a method for pattern matching (even just wildcards). The [docs](https://cloud.google.com/compute/docs/reference/rest/v1/subnetworks/list#query-parameters) say `The value must be a string, a number, or a boolean. The comparison operator must be either =, !=, >, or <.`